lib/ToolbarPanelHub.jsm

```diff
@@ -137,23 +137,45 @@ class _ToolbarPanelHub {
     });
   }
 
+  // Newer messages first and use `order` field to decide between messages
+  // with the same timestamp
+  _sortWhatsNewMessages(m1, m2) {
+    // Sort by published_date in descending order.
+    if (m1.content.published_date === m2.content.published_date) {
+      // Ascending order
+      return m1.order - m2.order;
```

Comment:
Do we want to make `order` required? Otherwise some undefined/NaN sorting behavior

Reply:
I think it's simpler to make it required. I'll update the PR.